zapcore/encoder.go

```diff
@@ -36,18 +64,45 @@ func LowercaseLevelEncoder(l Level, enc PrimitiveArrayEncoder) {
 	enc.AppendString(l.String())
 }
 
+// ColoredLowercaseLevelEncoder serializes a Level to a lowercase string and adds coloring.
+// For example, InfoLevel is serialized to "info".
+func ColoredLowercaseLevelEncoder(l Level, enc PrimitiveArrayEncoder) {
+	encodeWithColor(l, enc, l.String())
+}
+
 // CapitalLevelEncoder serializes a Level to an all-caps string. For example,
 // InfoLevel is serialized to "INFO".
 func CapitalLevelEncoder(l Level, enc PrimitiveArrayEncoder) {
 	enc.AppendString(strings.ToUpper(l.String()))
 }
 
+// ColoredCapitalLevelEncoder serializes a Level to an all-caps string and adds color.
+// For example, InfoLevel is serialized to "INFO".
+func ColoredCapitalLevelEncoder(l Level, enc PrimitiveArrayEncoder) {
+	encodeWithColor(l, enc, strings.ToUpper(l.String()))
+}
+
+func encodeWithColor(l Level, enc PrimitiveArrayEncoder, s string) {
+	colorCode, ok := levelToColorCode[l]
+	if !ok {
+		enc.AppendString(s)
+		return
+	}
+	enc.AppendString(fmt.Sprintf("\x1b[%dm%s\x1b[0m", colorCode, s))
+}
```

Comment:
Can we move the coloring logic (color codes, cached strings, etc.) to a separate file? This one's getting a bit big.

If we start coloring other output, we should make this a proper internal package.